testbeam_analysis/result_analysis.py

```diff
@@ -1299,3 +1299,273 @@ def histogram_track_angle(input_tracks_file, input_alignment_file=None, output_t
+
+                    intersection_x_cs_2_local, intersection_y_cs_2_local, intersection_z_cs_2_local = geometry_utils.apply_alignment(intersection_x_cs_2, intersection_y_cs_2, intersection_z_cs_2,
+                                                                                                                                     dut_index=actual_dut,
+                                                                                                                                     alignment=alignment,
+                                                                                                                                     inverse=True)
+
+                    intersection_x_cs_3_local, intersection_y_cs_3_local, intersection_z_cs_3_local = geometry_utils.apply_alignment(intersection_x_cs_3, intersection_y_cs_3, intersection_z_cs_3,
+                                                                                                                                     dut_index=actual_dut,
+                                                                                                                                     alignment=alignment,
+                                                                                                                                     inverse=True)
+
+                    intersection_x_cs_4_local, intersection_y_cs_4_local, intersection_z_cs_4_local = geometry_utils.apply_alignment(intersection_x_cs_4, intersection_y_cs_4, intersection_z_cs_4,
+                                                                                                                                     dut_index=actual_dut,
+                                                                                                                                     alignment=alignment,
+                                                                                                                                     inverse=True)
+
+                if not np.allclose(intersection_z_cs_4_local[np.isfinite(intersection_z_cs_4_local)], 0.0):
+                    print intersection_z_cs_4_local
+                    raise RuntimeError("Transformation into local coordinate system gives z != 0")
+
+                # project track intersections onto one pixel in order to increase statistics
+                projection_x_1 = np.mod(intersection_x_cs_1_local, np.array([actual_pixel_size[0]] * len(intersection_x_cs_1_local)))
+                projection_y_1 = np.mod(intersection_y_cs_1_local, np.array([actual_pixel_size[1]] * len(intersection_y_cs_1_local)))
+                projection_x_2 = np.mod(intersection_x_cs_2_local, np.array([actual_pixel_size[0]] * len(intersection_x_cs_2_local)))
+                projection_y_2 = np.mod(intersection_y_cs_2_local, np.array([actual_pixel_size[1]] * len(intersection_y_cs_2_local)))
```

Comment:
Too much copy paste is a sign that it is time for a for loop and/or helper function